p2p/host/hostv2/hostv2.go

```diff
@@ -87,6 +87,9 @@ func (r *GroupReceiverImpl) Close() error {
 func (r *GroupReceiverImpl) Receive(ctx context.Context) (
 	msg []byte, sender libp2p_peer.ID, err error,
 ) {
+	if r.sub == nil {
+		return nil, libp2p_peer.ID(""), fmt.Errorf("Receive: r.sub == nil")
```

Comment:
Thanks for catching this.  `r.sub == nil` means that the receiver has been closed (see `Close()` above), so a more descriptive error message would be:
```suggestion
		return nil, libp2p_peer.ID(""), fmt.Errorf("GroupReceiver has been closed")
```